src/DotVVM.Framework/Compilation/Binding/ExpressionHelper.cs

```diff
@@ -252,6 +259,53 @@ private static MethodRecognitionResult TryCallMethod(MethodInfo method, Type[] t
             };
         }
 
+        private static Type ResolveGenericTypeFromGivenExpressions(int i, Type genericArgument, ParameterInfo[] parameters, Expression[] args)
```

Comment:
`int i` is a bit confusing due to the cycle - could you please rename it, or use the genericParameter itself?